### PR TITLE
#143 cant export csv

### PR DESCRIPTION
**Pull request by joshxg**
- Can export/email 50,000 logs
- This is ~2mb of data for me- still takes around 10seconds, but it.. works!

### CODE REVIEW

File: android/.project [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+			<id>1622026265925</id>
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
+				<arguments>node_modules|.git|__CREATED_BY_JAVA_LANGUAGE_SERVER__</arguments>
+			</matcher>
+		</filter>
+	</filteredResources>
+</projectDescription>
```
Comment (joshxg) [inline android/.project:28]: this kept coming up in my git changes and wouldn't go away, maybe I shoulda just git ignored it 🤔

File: src/common/services/FormatService/FormatService.ts [hunk 2]
```diff
@@ -37,10 +38,21 @@ export class FormatService {
     return moment(date).format('DD MMMM');
   };
 
+  fileNameDate = (date: number): string => {
```
Comment (joshxg) [inline src/common/services/FormatService/FormatService.ts:41]: 🤔

File: src/features/Report/ReportManager.ts [hunk 3]
```diff
@@ -1,90 +1,27 @@
...
-  )
-
-select max(temperature) "Max Temperature", min(temperature) "Min Temperature",
-cbf.hasHotCumulative + cbf.hasColdCumulative as "Number of cumulative breaches",
-(select count(*) from temperaturebreach where sensorid = ?) as "Number of continuous breaches"
-from temperaturelog
-left join cumulativeBreachFields cbf
-where sensorid = ?
-group by sensorid`;
+import { STATS } from '~features/Report/queries';
```
Comment (mark-prins) [inline src/features/Report/ReportManager.ts:24]: could be added to line 3

File: src/ui/presentation/typography/NormalText.tsx [hunk 4]
```diff
@@ -1,10 +1,10 @@
 import React, { FC } from 'react';
-import { Text, TextStyle } from 'react-native';
-import { FONT, COLOUR } from '../../../common/constants';
+import { Text, TextStyle, TextProps } from 'react-native';
+import { FONT, COLOUR } from '~common/constants';
 
 const defaultStyle = { fontSize: FONT.SIZE.MS, fontFamily: FONT.FAMILY.REGULAR };
 
-interface NormalTextProps {
+interface NormalTextProps extends TextProps {
```
Comment (joshxg) [inline src/ui/presentation/typography/NormalText.tsx:7]: just changed to pass through base component `Text` props

File: src/ui/components/modal/ExportDataModal.tsx [hunk 5]
```diff
@@ -65,36 +68,37 @@ export const ExportDataModal: FC<ExportDataModalProps> = ({
...
                   value={values.username}
-                  label="Username"
-                  placeholder="Enter your name"
+                  label={t('USERNAME')}
+                  placeholder={t('ENTER_YOUR_USERNAME')}
                   onChangeText={e => {
                     handleChange('username')(e);
                   }}
-                  inputContainerStyle={{ width: width * 0.5 }}
+                  inputContainerStyle={{ width: width * 0.8 }}
```
Comment (mark-prins) [inline src/ui/components/modal/ExportDataModal.tsx:84]: could the magic number be moved to the constants?

File: src/ui/components/modal/ExportDataModal.tsx [hunk 6]
```diff
@@ -27,20 +28,22 @@ export const ExportDataModal: FC<ExportDataModalProps> = ({
   onClose,
   variant = 'export',
 }) => {
+  const formatter = useFormatter();
```
Comment (joshxg) [inline src/ui/components/modal/ExportDataModal.tsx:31]: Changed this component to show the date range for which the report is being generated, added translations and passing `from`/`to` variables to the actions

File: src/features/SensorDetail/Detail/DetailSlice.ts [hunk 7]
```diff
@@ -110,9 +110,12 @@ const selectFrom = ({ sensorDetail: { detail } }: RootState): number => {
   return from;
 };
 
-const fromTo = createSelector([selectFrom, selectTo], (from, to) => {
-  return { from, to };
-});
+const fromTo = createSelector<RootState, number, number, { from: number; to: number }>(
```
Comment (joshxg) [inline src/features/SensorDetail/Detail/DetailSlice.ts:113]: type update

File: src/features/Report/types.ts [hunk 8]
```diff
@@ -0,0 +1,109 @@
+export type ReportHeader =
```
Comment (joshxg) [inline src/features/Report/types.ts:1]: exported types

File: src/features/Report/queries.ts [hunk 9]
```diff
@@ -0,0 +1,91 @@
+export const BREACH_CONFIG_REPORT = `
```
Comment (joshxg) [inline src/features/Report/queries.ts:1]: exported the queries

File: src/features/Report/ReportManager.ts [hunk 10]
```diff
@@ -1,90 +1,27 @@
+import { FormatService } from './../../common/services/FormatService/FormatService';
```
Comment (joshxg) [inline src/features/Report/ReportManager.ts:1]: Refactored a little - originally the email function was just a copy paste of the 'write to disk' function which had 'hey also email' at the end. Added types.

Main change is using `parseAsync` and chunking the logs up to parse them all in smaller chunks. Doing one big parse makes it no work .. 🤷

**Review by mark-prins — APPROVED** [review]
Code looks good to me
I don't have 50,000+ records to export, so my tests are not very comprehensive. Seems to work though!

👍

**Comment by joshxg** [comment]
Oh .. those comments won't be too helpful for your review 🤣

**Comment by mark-prins** [comment]
Should the log generation work? Would that be helpful in testing this change?

I tried, with limited success...
generate-logs

File: src/common/services/FormatService/FormatService.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshxg) [reply src/common/services/FormatService/FormatService.ts]: a2133b0

Thanks - hardest part 😢

File: src/ui/components/modal/ExportDataModal.tsx (hunk 5 above)
Comment (joshxg) [reply src/ui/components/modal/ExportDataModal.tsx]: thanks. 
cdf8025 

I don't think I like this way of styling hehe

**Pull request merged**